builder/config_reader.go

```diff
@@ -124,16 +124,10 @@ func parseConfig(reader io.Reader, relativeToDir, path string) (Config, error) {
 
 	undecodedKeys := tomlMetadata.Undecoded()
 	if len(undecodedKeys) > 0 {
-		errorKeys := config.ParseUndecodedKeys(undecodedKeys)
+		unknownElementsMsg := config.FormatUndecodedKeys(undecodedKeys)
 
-		pluralizedElement := "element"
-		if len(errorKeys) > 1 {
```

Comment:
`ParseUndecodedKeys` returns a `string`, so `len(errorKeys) > 1` will (almost) always be true. This leads to awkward error messages like `unknown configuration elements 'thing'`.